modules/templates/handler.rb

```diff
@@ -7,23 +7,28 @@ module Proxy::Templates
   class Handler < ::Proxy::HttpRequest::ForemanRequest
     include Proxy::Log
 
-    def get_template(kind, token, static = false)
-      opts = { :token => token,
-               :url   => Proxy::Templates::Plugin.settings.template_url
-      }
-      opts[:static] = static if static
-      request = request_factory.create_get("/unattended/#{kind}", opts)
-      logger.debug "retrieving a template from %s/%s" % [Proxy::Templates::Plugin.settings.template_url, request.path]
-      res = send_request(request)
+    def get_template(kind, request, params)
+      url = Proxy::Templates::Plugin.settings.template_url
+      proxy_ip = URI.parse(url).host
+      opts = params.clone.merge(:url => url)
```

Comment:
Yes these are generated by Sinatra. Since I want this to be generic, I want to copy all the parameters except these. Not sure if there is more clever method than this one.